packages/mdc-card/mdc-card.scss

```diff
@@ -15,174 +15,151 @@
 //
 
 @import "@material/elevation/mixins";
+@import "@material/icon-toggle/mixins";
 @import "@material/theme/mixins";
-@import "@material/typography/mixins";
+@import "@material/ripple/mixins";
 @import "@material/rtl/mixins";
+@import "./mixins";
 
 // postcss-bem-linter: define card
+
 .mdc-card {
   @include mdc-elevation(2);
+  @include mdc-card-fill-color(background);
+  @include mdc-card-corner-radius(2px);
 
   display: flex;
   flex-direction: column;
-  justify-content: flex-end;
-  box-sizing: border-box;
-  padding: 0;
-  border-radius: 2px;
-  overflow: hidden;
+}
 
-  &__primary {
-    padding: 16px;
+//
+// Media
+//
 
-    // Add extra space before large title.
-    .mdc-card__title--large {
-      padding-top: 8px;
-    }
+.mdc-card__media {
+  position: relative;
+  background-repeat: no-repeat;
+  background-size: cover;
+}
 
-    // Add extra padding if this is the last block.
-    &:last-child {
-      padding-bottom: 24px;
-    }
-  }
+.mdc-card__media:first-child {
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+}
 
-  &__supporting-text {
-    @include mdc-typography(body1);
-    @include mdc-theme-prop(color, text-primary-on-light);
+.mdc-card__media:last-child {
+  border-bottom-left-radius: inherit;
+  border-bottom-right-radius: inherit;
+}
 
-    box-sizing: border-box;
-    padding: 8px 16px;
+.mdc-card__media--square {
+  @include mdc-card-media-aspect-ratio(1, 1);
+  @include mdc-card-media-aspect-ratio-base_;
```

Comment:
Shouldn't `...-aspect-ratio-base_` actually be included under just `.mdc-card__media` instead of in each of these modifiers? Otherwise how would users properly customize this without also applying one of these two modifiers first in order to get the base styles (since the base mixin is private)?

There is also no example of this in the demo page to prove that the exposed mixin is sufficient.

On the other hand, if we really only want people to use these two ratios, then we shouldn't expose the aspect-ratio mixin at all.

Reply:
Done